src/move/emojicoin_dot_fun/sources/tests.move

```diff
@@ -2218,114 +2297,601 @@
+            global_state,
+            vector::pop_back(&mut global_state_events),
+        );
+    }
+
+    #[test] fun period_times() {
+        let ms_per_s = get_MICROSECONDS_PER_SECOND();
+        assert!(get_PERIOD_1M() == 60 * ms_per_s, 0);
+        assert!(get_PERIOD_5M() == 5 * 60 * ms_per_s, 0);
+        assert!(get_PERIOD_15M() == 15 * 60 * ms_per_s, 0);
+        assert!(get_PERIOD_30M() == 30 * 60 * ms_per_s, 0);
+        assert!(get_PERIOD_1H() == 60 * 60 * ms_per_s, 0);
+        assert!(get_PERIOD_4H() == 4 * 60 * 60 * ms_per_s, 0);
+        assert!(get_PERIOD_1D() == 24 * 60 * 60 * ms_per_s, 0);
+    }
+
+    #[test, expected_failure(
+        abort_code = emojicoin_dot_fun::emojicoin_dot_fun::E_NO_MARKET,
+        location = emojicoin_dot_fun::emojicoin_dot_fun,
+    )] fun provide_liquidity_no_market() {
+        provide_liquidity<BlackCatEmojicoin, BlackCatEmojicoinLP>(
+            &get_signer(USER),
+            @black_cat_market,
+            1,
+        );
+    }
+
+    #[test, expected_failure(
+        abort_code = emojicoin_dot_fun::emojicoin_dot_fun::E_LIQUIDITY_NO_QUOTE,
+        location = emojicoin_dot_fun::emojicoin_dot_fun,
+    )] fun provide_liquidity_no_quote() {
+        init_package_then_exact_transition();
+        provide_liquidity<BlackCatEmojicoin, BlackCatEmojicoinLP>(
+            &get_signer(USER),
+            @black_cat_market,
+            0,
+        );
+    }
+
+    #[test, expected_failure(
+        abort_code = emojicoin_dot_fun::emojicoin_dot_fun::E_STILL_IN_BONDING_CURVE,
+        location = emojicoin_dot_fun::emojicoin_dot_fun,
+    )] fun provide_liquidity_still_in_bonding_curve() {
+        init_package_then_simple_buy();
+        provide_liquidity<BlackCatEmojicoin, BlackCatEmojicoinLP>(
+            &get_signer(USER),
+            @black_cat_market,
+            1
+        );
+    }
+
+    #[test] fun provide_liquidity_with_truncation() {
+        // Invoke an exact transition, then have simple buy user buy against the CPAMM.
+        init_package_then_exact_transition();
+        timestamp::update_global_time_for_test(EXACT_TRANSITION_TIME + 1);
+        mint_aptos_coin_to(SIMPLE_BUY_USER, SIMPLE_BUY_INPUT_AMOUNT);
+        swap<BlackCatEmojicoin, BlackCatEmojicoinLP>(
+            &get_signer(SIMPLE_BUY_USER),
+            @black_cat_market,
+            SIMPLE_BUY_INPUT_AMOUNT,
+            SWAP_BUY,
+            SIMPLE_BUY_INTEGRATOR,
+            SIMPLE_BUY_INTEGRATOR_FEE_RATE_BPS,
+        );
+
+        // Get the new CPAMM reserves.
+        let  ( _, _, _, cpamm_real_reserves, _, _, _, _, _, _, _, _, _) = unpack_market_view(
+            market_view<BlackCatEmojicoin, BlackCatEmojicoinLP>(@black_cat_market)
+        );
+        let (base, quote) = unpack_reserves(cpamm_real_reserves);
+        let cpamm_real_reserves = MockReserves { base, quote };
+
+        // Declare new amount of quote to provide.
+        let quote_amount = SIMPLE_BUY_INPUT_AMOUNT / 10;
```

Comment:
Is the `10` an arbitrary value or is it derived from something in the setup or a formula?

To be clear, I'm asking most of these questions because I use the unit tests pretty extensively to follow the logical flow of the contract 😅

Reply:
Addressed in b4f5b4d